src/main/java/org/example/examplefinalProject/service/ClassRoomService.java

```diff
@@ -1,51 +1,58 @@
 package org.example.examplefinalProject.service;
 
+import lombok.RequiredArgsConstructor;
 import org.example.examplefinalProject.entity.ClassRoom;
-import org.example.examplefinalProject.entity.Student;
+import org.example.examplefinalProject.entity.Teacher;
+import org.example.examplefinalProject.exception.InvalidParamsException;
 import org.example.examplefinalProject.repository.ClassRoomRepository;
 import org.example.examplefinalProject.repository.StudentRepository;
+import org.example.examplefinalProject.repository.TeacherRepository;
 import org.springframework.stereotype.Service;
 
+import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 
+
+@RequiredArgsConstructor
 @Service
 public class ClassRoomService {
+
     private final ClassRoomRepository classRoomRepository;
     private final StudentRepository studentRepository;
+    private final TeacherRepository teacherRepository;
+    private final TeacherService teacherService;
 
-    public ClassRoomService(ClassRoomRepository classRoomRepository, StudentRepository studentRepository) {
-        this.classRoomRepository = classRoomRepository;
-        this.studentRepository = studentRepository;
-    }
-
-    /**
-     * Vrací seznam všech tříd.
-     *
-     * @return Seznam tříd.
-     */
     public List<ClassRoom> findAll() {
         return classRoomRepository.findAll();
     }
 
-    /**
-     * Vrací detail třídy.
-     *
-     // @param id Identifikátor třídy.
-     * @return Detail třídy nebo {@code null}.
-     */
     public ClassRoom findById(int classRoomId) {
         return classRoomRepository.findById(classRoomId).orElse(null);
     }
 
-    /**
-     * Vrací počet studentů v dané třídě.
-     *
-     // @param id Identifikátor třídy.
-     * @return Počet studentů.
-     */
     public int studentCountById(int classRoomId) {
         return studentRepository.countByClassRoomClassRoomId(classRoomId);
     }
 
-
+    public void createClassRoom(String classRoomName, String teacherFullName) {
+        if (classRoomName == null) {
+            throw new InvalidParamsException("ClassRoomName is null ");
+        }
+
+        Optional<Teacher> teacher = Optional.empty();
+        String[] teacherNameParts = teacherFullName.split(" ");
```

Comment:
What about these names?
```
Ing Ucitel Novak
Mgr Janka Ucitelova Novakova
Beznemeno Beznepriezvisko
Ing Jozko Kovac Novotny csc.
Bc. Pavel Baláž, MBA
doc. PhDr. Aneta Dostálová, Ph.D., MBA
```

I know it is not common names just asking

Reply:
ano učitel vyplňuje pouze jméno a příjmení, tituly jsem zatím neřešila